src/multikey.rs

```diff
@@ -173,32 +170,18 @@ impl Multisecret {
         let tail =
             skip_prefix(suffix, ED25519_SUFFIX).ok_or_else(|| DecodeLegacyError::UnknownSuffix)?;
 
-        if data.len() != SecretKey::SIZE {
-            return Err(DecodeLegacyError::Ed25519WrongSize);
-        }
+        let tail_str = std::str::from_utf8(data).map_err(|_| DecodeLegacyError::InvalidUTF8)?;
 
-        if data[SecretKey::SIZE - 2] == b"="[0] {
-            return Err(DecodeLegacyError::Ed25519WrongSize);
-        }
+        let key_pair =
+            Keypair::from_base64(&tail_str).ok_or_else(|| DecodeLegacyError::InvalidBase64)?;
 
-        if data[SecretKey::SIZE - 1] != b"="[0] {
-            return Err(DecodeLegacyError::Ed25519WrongSize);
-        }
-
-        let mut dec_data = [0u8; 32];
-
-        base64::decode_config_slice(data, base64::STANDARD, &mut dec_data)
-            .map_err(|e| DecodeLegacyError::InvalidBase64(e))
-            .map(|_| {
-                let sk: SecretKey = SecretKey(dec_data);
-                (Multisecret(sk.to_owned()), tail)
-            })
+        Ok((Multisecret(key_pair), tail))
     }
 
     /// Serialize a `Multisecret` into a writer, using the
     /// [legacy encoding](https://spec.scuttlebutt.nz/datatypes.html#multikey-legacy-encoding).
     pub fn to_legacy<W: Write>(&self, w: &mut W) -> Result<(), io::Error> {
-        let data = base64::encode_config(&self.0.0[..], base64::STANDARD);
+        let data = base64::encode_config(&self.0.secret.0, base64::STANDARD);
```

Comment:
This one has to use Keypair::as_base64 to match js behavior (base64(pub + sec)).

Reply:
Done